### DISCUSSION TITLE
Welcome to ngx-rust Discussions!

### DISCUSSION THREAD

**Discussion by dekobon** [nginxinc/ngx-rust#23] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by WENPIN1** [comment]
How do we get the HTTP STATUS CODE in a module ?
It's unable to retrieve status code in request.headers_out_iterator().
Does it have any trick to get 'HTTP STATUS CODE' in an nginx module ?

Thank you!

-Patrick

**Reply by ivanitskiy** [reply]
I maybe don't fully understand the question. I will try to answer here.

HTTP Status code is an attribute of the response. The request has no status code as the request is "incoming" data. while a response structure is what will be sent back to the client.

From the [nginx dev guide](https://nginx.org/en/docs/dev/development_guide.html#http_response) I can see an example of a response handler.
      r->headers_out.status = NGX_HTTP_OK;

Please take a look here: https://github.com/nginxinc/ngx-rust/blob/ea23a229cfb4a2cd32a8238378e2780b062646dd/src/http/request.rs#L223

a curl example demonstrates how to set HTTP status code in a handler. https://github.com/nginxinc/ngx-rust/blob/ea23a229cfb4a2cd32a8238378e2780b062646dd/examples/curl.rs#L106 

ngx-rust is still an experimental module and requires writing "C in Rust".

**Reply by WENPIN1** [reply]
Thanks for your reply!

Finally, got a approach to get the http_staus code base on 
1. register module on NGX_HTTP_LOG_PHASE.
2. convert Request into() *const request_t pointer (a type alias of request_s)
and use request_s through ffi to get HTTP_STATUS in a similar logic of ‘ngx_http_log_status(ngx_http_request_t *r, u_char *buf, ngx_http_log_op_t *op)’

https://github.com/nginx/nginx/blob/ef96f5835468ff8d40df29b0ddbc04ec1e5e1582/src/http/modules/ngx_http_log_module.c#L855

A brute force approach.


Thank you again!

-WENPIN
